### PR TITLE
Update profile types

### PR DESCRIPTION
**Pull request by dbenamydd**
### What does this PR do?
See commit messages for details.

### Preview
https://docs-staging.datadoghq.com/dbenamy/profile-types/tracing/profiler/search_profiles/

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/tracing/profiler/search_profiles.md
Comment (sanchda) [inline content/en/tracing/profiler/search_profiles.md:214]: This is a bit confusing, since the native profiler will surface the internals of the Ruby runtime, but that's slightly different from what a user would think of as "Ruby code."  Is it clearer if we don't qualify the statement by just saying "Shows the time each function spent running on the CPU?"

File: content/en/tracing/profiler/search_profiles.md
Comment (sanchda) [inline content/en/tracing/profiler/search_profiles.md:185]: It seems a little odd to say that a function is _running_ if it is sleeping, but a walltime profiler would in fact attribute sleeps.

Is it more precise/comprehensible to say that the "function context is active?"  Something like that?

Total nit, please ignore if nothing good comes to mind.

File: content/en/tracing/profiler/search_profiles.md
Comment (dbenamydd) [reply content/en/tracing/profiler/search_profiles.md:214]: Nice catch, bad copy/pasta.

File: content/en/tracing/profiler/search_profiles.md
Comment (dbenamydd) [reply content/en/tracing/profiler/search_profiles.md:185]: "Function context is active" seems a bit complex. I'm open to other ideas but don't have anything myself yet.

File: content/en/tracing/profiler/search_profiles.md
Comment (felixge) [inline content/en/tracing/profiler/search_profiles.md:143]: > and total memory usage

NIT: I think this addition would be a better fit for the `Heap Live Size` profile. Allocation rate and memory usage are not always correlated IMO.

File: content/en/tracing/profiler/search_profiles.md
Comment (felixge) [inline content/en/tracing/profiler/search_profiles.md:157]: NIT: I'm okay with renaming this, but we should probably do the same in the UI if we change it in the docs?

image

File: content/en/tracing/profiler/search_profiles.md
Comment (felixge) [inline content/en/tracing/profiler/search_profiles.md:161]: ⚠️ This is not GA. We decided to put it on hold because the overhead is too high.

**Review by felixge — CHANGES_REQUESTED** [review]
Reviewed the Go changes. PTAL at my nits as well as one ⚠️ that we should fix before merging this.

File: content/en/tracing/profiler/search_profiles.md
Comment (dbenamydd) [reply content/en/tracing/profiler/search_profiles.md:185]: Thanks for the suggestion :-) It could be doing things other than those 2 too.

File: content/en/tracing/profiler/search_profiles.md
Comment (dbenamydd) [reply content/en/tracing/profiler/search_profiles.md:143]: Whoops! Bad edit, nice catch :-)

File: content/en/tracing/profiler/search_profiles.md
Comment (dbenamydd) [reply content/en/tracing/profiler/search_profiles.md:157]: Good call. Many but not all profile types are singular so I was trying to be more consistent there, but let me roll this back for now since I don't have time to update everything.

File: content/en/tracing/profiler/search_profiles.md
Comment (dbenamydd) [reply content/en/tracing/profiler/search_profiles.md:161]: Un-adding!

File: content/en/tracing/profiler/search_profiles.md
Comment (morrisonlevi) [inline content/en/tracing/profiler/search_profiles.md:210]: We have an env var `DD_PROFILING_EXPERIMENTAL_CPU_ENABLED` which controls whether CPU Time is gathered. It's biased towards I/O functions but it's cheap. Not sure if it's worth mentioning something here, or if we should wait.

Comment (dbenamydd) [reply content/en/tracing/profiler/search_profiles.md:210]: Let's wait till it's on by default.

File: content/en/tracing/profiler/search_profiles.md
Comment (Qard) [reply content/en/tracing/profiler/search_profiles.md:188]: Just swap "function" for "application" maybe? 🤔

**Review by Qard — COMMENTED** [review]
Other than comment about wall time, LGTM.

File: content/en/tracing/profiler/search_profiles.md
Comment (dbenamydd) [reply content/en/tracing/profiler/search_profiles.md:188]: Thanks for the idea :-) That wouldn't be quite the same thing if I understand you correctly. It's saying the width of each frame/box/function shown on the flame graph is based on how much time happened while that specific function was "running" (for various values of running).

**Comment by dbenamydd** [comment]
@kayayarai All yours!

**Review by kayayarai — APPROVED** [review]

**Comment by dbenamydd** [comment]
Thanks!

**Pull request merged**
